### PR TITLE
sys/shell/vfs: make output of vfs df human readable

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Currently `vfs df` outputs size as filesystem blocks. It is not clear how big a filesystem block is, so the output is hard to interpret.

Instead, output the sizes in Bytes like `df -h`. 

### Testing procedure

Run e.g. `examples/filesystem`:

```
2022-09-04 12:46:51,962 # > vfs df
2022-09-04 12:46:51,967 # Mountpoint              Total         Used    Available     Capacity
2022-09-04 12:46:51,973 # /sd0                   15223M      129912k    15458440k       0%
2022-09-04 12:46:52,006 # /nvm0                     32M         220k       32548k       0%
2022-09-04 12:46:52,011 # /const                     27           27            0     100%
```

### CODE REVIEW

File: sys/shell/commands/sc_vfs.c [hunk 1]
```diff
@@ -137,6 +138,37 @@ static int _errno_string(int err, char *buf, size_t buflen)
...
+{
+    unsigned long len;
+    const char *unit;
+
+    if (size == 0) {
+        len = 0;
+        unit = NULL;
+    } else if ((size & (GiB(1) - 1)) == 0) {
+        len = size / GiB(1);
+        unit = "G";
```
Comment (maribu) [inline sys/shell/commands/sc_vfs.c:151]:
```suggestion
        unit = "GiB";
```

File: sys/shell/commands/sc_vfs.c [hunk 2]
```diff
@@ -137,6 +138,37 @@ static int _errno_string(int err, char *buf, size_t buflen)
...
+    }
+    else if ((size & (MiB(1) - 1)) == 0) {
+        len = size / MiB(1);
+        unit = "M";
```
Comment (maribu) [inline sys/shell/commands/sc_vfs.c:155]:
```suggestion
        unit = "MiB";
```

File: sys/shell/commands/sc_vfs.c [hunk 3]
```diff
@@ -137,6 +138,37 @@ static int _errno_string(int err, char *buf, size_t buflen)
...
+    }
+    else if ((size & (KiB(1) - 1)) == 0) {
+        len = size / KiB(1);
+        unit = "k";
```
Comment (maribu) [inline sys/shell/commands/sc_vfs.c:159]:
```suggestion
        unit = "KiB";
```

File: sys/shell/commands/sc_vfs.c [hunk 4]
```diff
@@ -137,6 +138,37 @@ static int _errno_string(int err, char *buf, size_t buflen)
...
+    } else {
+        len = size;
+        unit = NULL;
```
Comment (maribu) [inline sys/shell/commands/sc_vfs.c:162]:
```suggestion
        unit = "B";
```

File: sys/shell/commands/sc_vfs.c [hunk 5]
```diff
@@ -137,6 +138,37 @@ static int _errno_string(int err, char *buf, size_t buflen)
...
+    }
+
+    if (unit) {
```
Comment (maribu) [inline sys/shell/commands/sc_vfs.c:165]: If you accept the suggestion, unit will never be NULL

**Review by maribu — COMMENTED** [review]
Looks good to me.

Since you are touching the code anyway: Care to replace "Capacity" in the table header by something more sensible? E.g. when capacity is at 100%, I would assume an empty file storage. I think `df` is showing `Use%` there, if I recall correctly

File: sys/shell/commands/sc_vfs.c (hunk 5 above)
Comment (benpicco) [reply sys/shell/commands/sc_vfs.c:165]: This was done to get the alignment right - and is still needed since the suggested unit names differ in length

**Comment by benpicco** [comment]
Looks like this now

```
2022-09-07 22:09:16,902 # Mountpoint              Total         Used    Available     Use%
2022-09-07 22:09:16,908 # /sd0                15223 MiB   131288 kiB 15457064 kiB       0%
2022-09-07 22:09:16,940 # /nvm0                  32 MiB      220 kiB    32548 kiB       0%
2022-09-07 22:09:16,946 # /const                   27 B         27 B          0 B     100%
```

**Review by maribu — APPROVED** [review]
Thx :)

File: sys/shell/commands/sc_vfs.c [hunk 6]
```diff
@@ -137,6 +138,37 @@ static int _errno_string(int err, char *buf, size_t buflen)
...
+        len = size / GiB(1);
+        unit = "GiB";
+    }
+    else if ((size & (MiB(1) - 1)) == 0) {
+        len = size / MiB(1);
+        unit = "MiB";
+    }
+    else if ((size & (KiB(1) - 1)) == 0) {
+        len = size / KiB(1);
+        unit = "kiB";
```
Comment (maribu) [inline sys/shell/commands/sc_vfs.c:159]:
```suggestion
        unit = "KiB";
```

Almost missed this. Please squash, ACK remains valid

**Pull request merged**